### ISSUE TITLE
Poll: what would you like to add to improve option --pretty?

### ISSUE THREAD

**Issue by genivia-inc** [Genivia/ugrep#52]
Option `--pretty` produces filename headings (`-+, --heading`) and tabbing (`-T, --initial-tab`) when the output is sent to a terminal. That is basically it.

In fact, `-T+` is even shorter to type than `--pretty`!

So this option isn't doing much and seems wasteful to include. But it has a lot of potential and we can improve it.

What would you like `--pretty` to do in addition to `-T+` when the output is sent to a terminal?

Here are some suggestions, but please add your own if you don't find this useful:
[ ] enable option `-n` to show line numbers
[ ] enable `--sort` to sort by filename, if no `--sort` option was specified
[ ] use different colors (or additional colors?) than the default

**Comment by genivia-inc** [comment]
The following update is planned to improve `--pretty`:
~~~
    --pretty
            When output is sent to a terminal, enables --color, --heading, -n,
            --sort and -T, if these options are not explicitly disabled or set.
~~~
Option `--pretty` shows line numbers. If that is really not desired, then `--no-line-number` can be used. If `--sort` is specified then `--pretty` will not override the request sort order.

**Comment by genivia-inc** [comment]
Implemented and tested in release v2.5.0 c990e6a.
